app/dao/template_categories_dao.py

```diff
@@ -69,12 +68,12 @@ def dao_delete_template_category_by_id(template_category_id, cascade=False):
                     if template.template_type == "sms"
                     else template_category.email_process_type
                 )
-                template.category = dao_get_template_category_by_id(default_category_id)
+                template.template_category_id = dao_get_template_category_by_id(default_category_id).id
                 template.updated_at = datetime.utcnow()
                 db.session.add(template)
+                db.session.commit()
 
         db.session.delete(template_category)
-        db.session.commit()
```

Comment:
Why did you move the commit?

Reply:
I removed it because with `@transactional` we auto commit after the method executes. Buuuut in testing I had removed `@transactional`. I'll add it back in.